src/Http/Http/src/Internal/BufferingHelper.cs

```diff
@@ -21,15 +22,17 @@ public static HttpRequest EnableRewind(this HttpRequest request, int bufferThres
             var body = request.Body;
             if (!body.CanSeek)
             {
-                var fileStream = new FileBufferingReadStream(body, bufferThreshold, bufferLimit, AspNetCoreTempDirectory.TempDirectoryFactory);
+                var httpBufferingOptions = request.HttpContext.RequestServices.GetRequiredService<IOptions<HttpBufferingOptions>>();
+                var factory = new HttpFileBufferingStreamFactory(httpBufferingOptions);
```

Comment:
Why isn't the factory being resolved from DI? Yes something would need to register it, probably Hosting with the rest of the default services.
https://github.com/aspnet/AspNetCore/blob/70b478136a702d2b6061ac9b159029189ad230d0/src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs#L84
https://github.com/aspnet/AspNetCore/blob/70b478136a702d2b6061ac9b159029189ad230d0/src/Hosting/Hosting/src/WebHostBuilder.cs#L284

Reply:
Great, I thought about register and resolve the factory but didn't know where to register it. Hosting sounds good, thanks :)

I can register together the IPostConfiguration for the options and the factory, probably with an Extension method, like AddHttpBuffering()?

The scope I believe it should be a singleton, out of curiosity, why the `DefaultHttpContextFactory` in the Generic is a Singleton and in the other one is Transient? Should I follow same pattern here?

Reply:
> I can register together the IPostConfiguration for the options and the factory, probably with an Extension method, like AddHttpBuffering()?

Sounds good.

> The scope I believe it should be a singleton, out of curiosity, why the DefaultHttpContextFactory in the Generic is a Singleton and in the other one is Transient? Should I follow same pattern here?

I don't recall, but follow what we do in Generic, it's more recent.